corehq/apps/app_manager/suite_xml.py

```diff
@@ -226,11 +226,73 @@ def set_display(self, locale_id=None, media_image=None, media_audio=None):
             self.text = text
 
 
-class Command(DisplayNode, IdNode):
+class LocaleId(XmlObject):
+    ROOT_NAME = 'locale'
+    locale_id = StringField('@id')
+
+
+class MediaText(XmlObject):
+    ROOT_NAME = 'text'
+    form_name = StringField('@form', choices=['image', 'audio'])  # Nothing XForm-y about this 'form'
+    locale = NodeField('locale', LocaleId)
+
+
+class LocalizedMediaDisplay(XmlObject):
+    ROOT_NAME = 'display'
+
+    media_text = NodeListField('text', MediaText)
+
+
+class TextOrDisplay(XmlObject):
+    text = NodeField('text', Text)
+    display = NodeField('display', LocalizedMediaDisplay)
+
+    def __init__(self, node=None, context=None,
+                 menu_locale_id=None, image_locale_id=None, audio_locale_id=None,
+                 media_image=None, media_audio=None, **kwargs):
+        super(TextOrDisplay, self).__init__(node, context, **kwargs)
+        text = Text(locale_id=menu_locale_id) if menu_locale_id else None
+
+        media_text = []
+        if media_image:
+            media_text.append(MediaText(
+                locale=LocaleId(locale_id=image_locale_id),
+                form_name='image',
+            ))
+        if media_audio:
+            media_text.append(MediaText(
+                locale=LocaleId(locale_id=audio_locale_id),
+                form_name='audio'
+            ))
+
+        if media_text:
+            self.display = LocalizedMediaDisplay(
+                text=text,
+                media_text=[text] + media_text if text else [] + media_text
```

Comment:
no need for the empty list at the end: `media_text=[text] + media_text if text else media_text`